core/src/test/scala/unit/kafka/admin/AclCommandTest.scala

```diff
@@ -0,0 +1,104 @@
+    Resource.ClusterResource -> Array("--cluster"),
+    new Resource(ConsumerGroup, "test") -> Array("--consumer-group", "test")
+  )
+
+  private val operations: Set[Operation] = Set(Read, Write)
+  private val operationsString = operations.mkString(AclCommand.delimeter)
+  private val users = Set(KafkaPrincipal.fromString("User:test1"), KafkaPrincipal.fromString("User:test2"))
+  private val usersString = users.mkString(AclCommand.delimeter)
+  private val hosts = Set("host1", "host2")
+  private val hostsString = hosts.mkString(AclCommand.delimeter)
+
+  @Test
+  def testAclCli() {
+    val brokerProps: Properties = TestUtils.createBrokerConfig(0, zkConnect)
+    brokerProps.put(KafkaConfig.AuthorizerClassNameProp, "kafka.security.auth.SimpleAclAuthorizer")
+
+    //create a temp server.properties file
+    val config: File = TestUtils.tempFile()
+    brokerProps.store(new FileWriter(config), "test-broker-config")
+
+    val AllowAclsAndCommandOptions = getAclToCommand(Allow)
+    val DenyAclsAndCommandOptions = getAclToCommand(Deny)
+    val AllowAndDenyAclsAndCommandOptions = (AllowAclsAndCommandOptions._1 ++ DenyAclsAndCommandOptions._1,
+      AllowAclsAndCommandOptions._2 ++ DenyAclsAndCommandOptions._2)
+    val aclsToCmd = Set(AllowAclsAndCommandOptions, DenyAclsAndCommandOptions, AllowAndDenyAclsAndCommandOptions)
+
+    val args: Array[String] = Array("--config", config.getPath,"--operations", operationsString)
+
+    //now test add acls for each type of resource and with only allow, only deny and both allow and deny options.
+    for((resource, resourceCmd) <- resourceToCommand) {
+      for((acls, cmd) <- aclsToCmd) {
+        AclCommand.main(args ++ cmd ++ resourceCmd :+ "--add")
+        Assert.assertEquals(acls, getAuthorizer(brokerProps).getAcls(resource))
+
+        //cleanup
+        getAuthorizer(brokerProps).removeAcls(resource)
+        Assert.assertEquals(Set.empty[Acl], getAuthorizer(brokerProps).getAcls(resource))
+      }
+    }
+
+    //The cleanup could have been done using --remove option instead of calling removeAcls directly on authorizer
+    //but System.setIn and Console.setIn both do not work in scala-2.11 so this test gets stuck trying to read from stdin.
+    //val in = new ByteArrayInputStream("y".getBytes())
+    //System.setIn(in)
+    //AclCommand.main(args :+ "--remove")
+    //assertTrue(authorizer.getAcls(new Resource(Topic, topic)).isEmpty)
+  }
+
+  private def getAclToCommand(permissionType: PermissionType) : (Set[Acl], Array[String]) = {
+    (AclCommand.getAcls(users, permissionType, operations, hosts),
+      if(permissionType == Allow)
+        Array("--allowprincipals", usersString,
+        "--allowhosts", hostsString)
+      else
+        Array("--denyprincipals", usersString,
+          "--denyhosts", hostsString))
+  }
+
+  def getAuthorizer(props: Properties): Authorizer = {
+    val kafkaConfig = KafkaConfig.fromProps(props)
+    val authZ: SimpleAclAuthorizer = new SimpleAclAuthorizer
```

Comment:
Yes, that's OK to keep. There is a way to avoid the problem when defining `Read`, `Write`, etc. I'll explain in more detail when I have access to a computer instead of my phone. :)